main.go

```diff
@@ -191,9 +196,20 @@ func registerMetricsRouter(router *mux.Router) error {
 	return nil
 }
 
+// getHealthCheckURL - extracts the health check URL.
+func getHealthCheckURL(endpoint, healthCheckPath string, healthCheckPort int) string {
+	healthCheckURL, _ := url.Parse(strings.TrimSuffix(endpoint, slashSeparator) + healthCheckPath)
+	if healthCheckPort != 0 {
+		host, _, _ := net.SplitHostPort(healthCheckURL.Host)
+		healthCheckURL.Host = host + colonSeparator + strconv.Itoa(healthCheckPort)
```

Comment:
You should use net.JoinHostPort()

Reply:
Updated the code using `net.JoinHostPort()`